modules/afsocket/socket-options.c

```diff
@@ -76,6 +76,37 @@ socket_options_setup_socket_method(SocketOptions *self, gint fd, GSockAddr *bind
   return TRUE;
 }
 
+gboolean
+socket_options_set_sndbuf(SocketOptions *s, gint so_sndbuf)
+{
+  if (so_sndbuf < 0)
+    {
+      msg_error("Invalid value for so_sndbuf",
+                evt_tag_int("so_sndbuf", so_sndbuf),
+                evt_tag_int("minimum value", 0),
+                evt_tag_int("maximum value", G_MAXINT));
+      return FALSE;
+    }
```

Comment:
This should be reported by yyerror() and not by a separate log message.

There's a CHECK_ERROR() macro that makes this trivial from grammars.

Also, a number of tags use spaces instead of underscore (minimum value) above.

I would not merge this in this form.